editor/js/Loader.js

```diff
@@ -64,931 +64,856 @@ function Loader( editor ) {
 
 	this.loadFile = function ( file, manager ) {
 
-		const filename = file.name;
-		const extension = filename.split( '.' ).pop().toLowerCase();
+		const extension = file.name.split( '.' ).pop().toLowerCase();
 
 		const reader = new FileReader();
 		reader.addEventListener( 'progress', function ( event ) {
 
 			const size = '(' + editor.utils.formatNumber( Math.floor( event.total / 1000 ) ) + ' KB)';
 			const progress = Math.floor( ( event.loaded / event.total ) * 100 ) + '%';
 
-			console.log( 'Loading', filename, size, progress );
+			console.log( 'Loading', file.name, size, progress );
 
 		} );
 
-		switch ( extension ) {
-
-			case '3dm':
-
-			{
-
-				reader.addEventListener( 'load', async function ( event ) {
-
-					const contents = event.target.result;
+		if ( extension in fileHandlers ) {
 
-					const { Rhino3dmLoader } = await import( 'three/addons/loaders/3DMLoader.js' );
+			if ( extension === 'js' || extension === 'json' ) {
 
-					const loader = new Rhino3dmLoader();
-					loader.setLibraryPath( '../examples/jsm/libs/rhino3dm/' );
-					loader.parse( contents, function ( object ) {
+				fileHandlers[ extension ]( editor, manager, reader, file, this.texturePath );
```

Comment:
To make this a bit more readable, can you write it like so:
```js
const handleRequest = fileHandlers[ extension ];
```
And later in the `if` block:
```js
handleRequest( editor, manager, reader, file );
```

Reply:
ok